Source/WebCore/page/EventHandler.cpp

```diff
@@ -380,6 +380,20 @@ bool EventHandler::eventLoopHandleMouseDragged(const MouseEventWithHitTestResult
 
 #endif
 
+// Refetch the event target node if it is removed or currently is the shadow node inside an <input> element.
+// If a mouse event handler changes the input element type to one that has a widget associated,
+// we'd like to EventHandler::handleMousePressEvent to pass the event to the widget and thus the
+// event target node can't still be the shadow node.
+static inline bool shouldRefetchEventTarget(const MouseEventWithHitTestResults& mouseEvent)
+{
+    RefPtr targetNode = mouseEvent.targetNode();
+    ASSERT(targetNode);
+    RefPtr shadowRoot = dynamicDowncast<ShadowRoot>(*targetNode);
+    if (!targetNode->parentNode())
+        return true;
```

Comment:
I'd probably invert the order here, to prevent an unnecessary `dynamicDowncast` in the case where `targetNode` has no parent.

```suggestion
    if (!targetNode->parentNode())
        return true;
    RefPtr shadowRoot = dynamicDowncast<ShadowRoot>(*targetNode);
```

Reply:
Sure - I will update it. Thanks!

Reply:
Done.